src/code/actions.js

```diff
@@ -32,12 +32,13 @@ export function changeAuthoring(authoring) {
   };
 }
 
-export function navigateToChallenge(_case, challenge) {
+export function navigateToChallenge(level, _case, challenge) {
```

Comment:
One other thought: consider naming the outer object `routeSpec` which disentangles the name of the outer object from the terminology of its inner components.